### PR TITLE
Support ecma part 2

### PR DESCRIPTION
**Pull request by AtofStryker**
a follow up to #1 

Essentially cut over the the webpack loader to use ES6 Modules with working tests. Added a circle ci job to test to kick off on code pushes. README is updated to accurately reflect usage and requirements as well as package.json

Eventually I will make a PR to get this as a feature in the original library via a flag.

~I am also starting with a `beta` canary on publish (once I have access) to test functionality outside `npm link`~ . We will be referencing this in the webapp via a tag on `master`

### CODE REVIEW

File: package.json
Comment (AtofStryker) [inline package.json]: this will be v`1.0.0` once testing is finalized

**Review by jessewagstaff — APPROVED** [review]

File: README.md
Comment (electrachong) [inline README.md]:
```suggestion
The variable option is `required` to compile EJS templates into ES compatible modules. If the `variable` option is not provided as a loader or `query` [option](https://webpack.js.org/concepts/loaders/#loader-features), an `UnsupportedConfiguration` Exception will be thrown. Please see https://github.com/lodash/lodash/issues/3709#issuecomment-375898111 for additional details 
```

File: README.md
Comment (electrachong) [inline README.md]:
```suggestion
* 1.0.0 - Initial fork of [ejs-loader](https://github.com/difelice/ejs-loader) with support for ECMA Modules and removing CommonsJS
```

**Review by electrachong — APPROVED** [review]
If the original library rejects the pr, will we just need to patch any updates that we want to apply 
 to ejs-loader ourselves?

my contributions are just typos and spell check :P

**Comment by AtofStryker** [comment]
>If the original library rejects the pr, will we just need to patch any updates that we want to apply
to ejs-loader ourselves?

@electrachong unfortunately yes, but hopefully we can avoid that and not maintain this fork long term. Hopefully they will accept the change 😄 

> my contributions are just typos and spell check :P

Thank you as always for being my spellchecker 😆

**Comment by AtofStryker** [comment]
@electrachong spelling errors should now be fixed 😄

File: README.md
Comment (al-the-x) [inline README.md]: Whoops.

File: README.md
Comment (al-the-x) [inline README.md:4]: I think Travis-CI is free to use for OSS, including forks, so we could just change the namespace here to `groveco` and keep the `.travis.yml` basically unchanged... But good work converting to Circle CI!

File: README.md
Comment (al-the-x) [inline README.md]: Do not recommend. Use `0.4.0-esm` or something similar. Beta versions (`<1.y.z`) are allowed to break stuff in minor releases (`0.x`). The package maintainers should decide when the thing goes to `1.0.0`.

File: index.js
Comment (al-the-x) [inline index.js:15]: Recommended: extract magic string to property of `UnsupportedConfiguration` or a constant and use that in the test, too.

File: __tests__/index.test.js
Comment (al-the-x) [inline __tests__/index.test.js:41]: Consider testing the error type here instead of a message string?

**Review by al-the-x — CHANGES_REQUESTED** [review]

File: README.md
Comment (AtofStryker) [reply README.md:4]: @al-the-x thanks! Since we are maintaining this as an org temporarily (I hope), we probably want to keep CI providers consistent to prevent even more cognitive overhead. I think we should keep the circle config since the changes in this repo will not be contributed directly upstream. https://github.com/difelice/ejs-loader/pull/41 and https://github.com/difelice/ejs-loader/pull/42 are similar to what we have going on here, but took a little bit more work to be backwards compatible (feature flag with option to opt in).

File: __tests__/index.test.js
Comment (AtofStryker) [reply __tests__/index.test.js:41]: @al-the-x I wanted to originally do that but would have to export the error and didn't want to add complexity by exporting an ES6 class. I figured a test on a magic string was the next best option 🤷‍♂️

File: index.js
Comment (AtofStryker) [reply index.js:15]: same lines as https://github.com/groveco/ejs-loader/pull/2#discussion_r433501567. I'm good with doing it but not sure if the pain is worth the benefit of checking specific exception casting.

File: README.md
Comment (AtofStryker) [reply README.md]: @al-the-x I'm good with that. I wasn't sure how to version a non version-able release 😄 . The `0.4.0-esm` is just to announce on our end where the fork begins? FWIW these versions will not be what is dictated by the original package maintainers and have no bearing or effect on the actual PRs to the original repo.

Comment (racoleman) [reply README.md]: The original package is no longer under active development and has no owner, so I'm going to venture that it will never hit `1.0.0`. I'm also not optimistic about @AtofStryker's PRs being accepted. Until that happens, this is our fork and we _are_ the maintainers.

I made the change to `1.0.0` because the special allowance for beta versions forces us to pin the package or risk grove core breaking spontaneously, so we gain none of the benefits of semantic versioning. At `1.0.0`, we have the capability to release patches that will be pulled in automatically, while reserving breaking changes for major versions.

File: README.md
Comment (AtofStryker) [reply README.md]: @al-the-x updated with proper docs

**Comment by AtofStryker** [comment]
@jessewagstaff @electrachong @al-the-x @racoleman the changes needed here were implemented upstream. We likely don't need this fork anymore and can just use `ejs-loader` version 0.4.1 🎉 .

From https://github.com/groveco/grove/pull/9160

> I am now a contributor on ejs-loader and ES6 modules PR was merged 🎉 . update the webpack options parsing was closed due to some of the commits being included in the ES6 modules PR. However, the default options commit was missing, which was merged in right after

> ES6 modules was included in version 0.4.0 and default options commit was included in version 0.4.1. The webapp now has both 😄 .

**Pull request closed without merging**
